docs/source/user-guide/vision/quickstarts/plot_segmentation_tutorial.py

```diff
@@ -27,151 +27,108 @@
 #
-# The SegmentationData class contains additional data and general methods intended for easy access to relevant metadata
-# for semantic segmentation ML models validation.
-# To learn more about the expected format please visit the API reference for the
-# :class:`deepchecks.vision.segmentation_data.SegmentationData` class.
+# To learn more about the expected formats, please visit the
+# :doc:`supported tasks and formats guide </user-guide/vision/supported_tasks_and_formats>`
+#
+# First, we'll create the collate function that will be used by the DataLoader.
+# In pytorch, the collate function is used to transform the output batch to any custom format, and we'll use that
+# in order to transform the batch to the correct format for the checks.
 
+import torch
+import torchvision.transforms.functional as F
 
-class CocoSegmentationData(SegmentationData):
-    """Class for loading the COCO segmentation dataset, inherits from :class:`~deepchecks.vision.SegmentationData`.
+def deepchecks_collate_fn(batch):
+    """Return a batch of images, labels and predictions for a batch of data. The expected format is a dictionary with
+    the following keys: 'images', 'labels' and 'predictions', each value is in the predefined format for the task.
 
-    Implement the necessary methods to load the dataset.
+    To learn more about the expected formats, please visit the
+    :doc:supported tasks and formats guide </user-guide/vision/supported_tasks_and_formats>``
     """
+    # batch received as iterable of tuples of (image, label) and transformed to tuple of iterables of images and labels:
+    batch = tuple(zip(*batch))
 
-    def batch_to_labels(self, batch):
-        """Extract from the batch only the labels and return the labels in format (H, W).
-
-        See SegmentationData for more details on format.
-        """
-        return batch[1]
-
-    def infer_on_batch(self, batch, model, device):
-        """Infer on a batch of images and return predictions in format (C, H, W), where C is the class_id dimension.
-
-        See SegmentationData for more details on format.
-        """
-        normalized_batch = [F.normalize(img.unsqueeze(0).float() / 255,
-                                        mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]) for img in batch[0]]
-
-        predictions = [model(img)["out"].squeeze(0).detach() for img in normalized_batch]
-        predictions = [torch.nn.functional.softmax(pred, dim=0) for pred in predictions]
+    # images:
+    images = [tensor.numpy().transpose((1, 2, 0)) for tensor in batch[0]]
 
-        return predictions
+    #labels:
+    labels = batch[1]
 
-    def batch_to_images(self, batch):
-        """Convert the batch to a list of images, where each image is a 3D numpy array in the format (H, W, C)."""
-        return [tensor.numpy().transpose((1, 2, 0)) for tensor in batch[0]]
+    #predictions:
```

Comment:
maybe state what is the required format for predictions

Reply:
Discussed in person that no